haskell/actions.bzl

```diff
@@ -866,35 +862,41 @@ def _zencode(s):
   return s.replace("Z", "ZZ").replace("_", "ZU").replace("/", "ZS")
 
 def get_pkg_name(ctx):
-  """Get package name. Package name includes Bazel package and name of the
```

Comment:
You point out that funny package names won't just be visible in documentation, but much worse, in source code too (if using `-XPackageImports`). Funny names in documentation is less bad because if the names change, nothing breaks. Fortunately, we can separate package-id from package-name, it's just that underscores are still illegal.

If so then I'd lean towards banning underscores outright. That way no encoding necessary, no funny names in package imports. Either that or `s/_/-/`, which yes, is technically a non-injective encoding, but since package names don't need to be unique (only package-id's), that should be ok. And it lets users keep to Bazel-conventional package naming style if desired (i.e. underscores everywhere, no hyphens).